auth/README.md

```diff
@@ -0,0 +1,364 @@
+              error.reason.invalidParameters
+         }
+}
+
+```
+
+```java
+
+// Java
+authenticator.clientCredentials(new VimeoCallback<BasicAccessToken>() {
+
+    @Override
+    public void onError(@NotNull Error error) {
+        
+        if (error instanceof Error.Api) {
+
+            Api apiError = (Api) error;
+            ApiError reason = apiError.getReason();
+            
+            reason.getErrorCode();
+            reason.getErrorMessage();
+            reason.getInvalidParameters();
+            
+        }
+    }
+});
+
+```
+
+The error message and code could be extracted from the `ApiError` object.
+
+### Exception
+
+If any exception occurs during your request, you will also be informed through the `onError` callback method. A exception may occur if you don't have a connection and you had attempted to make a request. 
+
+
+```kotlin
+
+// Kotlin
+ override fun onError(error: VimeoResponse.Error) { 
+         
+        if (error is VimeoResponse.Error.Exception) {
+             error.exception.developerMessage
+             error.exception.printStackTrace()
+        }
+}
+
+```
+
+```java
+
+// Java
+authenticator.clientCredentials(new VimeoCallback<BasicAccessToken>() {
+
+    @Override
+    public void onError(@NotNull Error error) {
+                
+        if (error instanceof Error.Exception) {
+                                   
+            Error.Exception exception = ((Exception) error)
+            exception.getThrowable();
+            exception.getThrowable().printStackTrace();
+                
+        }
+    }
+});
+
+```
+
+You could extract information from the throwable that is returned to you.
+
+### Generic Error
+
+A generic error could occur if the SDK failed to parse the response from the API or the API didn't return a response at all. The request returned an http status code and no exceptions were thrown. 
+
+```kotlin
+
+// Kotlin
+ override fun onError(error: VimeoResponse.Error) { 
+
+       if(error is Error.Generic) {
+           error.generic.httpStatusCode
+           error.generic.rawResponse
+       }
+}
+```
+
+```java
+
+// Java
+@Override
+public void onError(@NotNull Error error) {
+                
+    if (error instanceof Error.Generic) {
+
+        Error.Generic generic = ((Generic) error);
+        generic.getHttpStatusCode();
+        generic.getRawResponse();
+
+    }                
+}
+
+```
+
+The HTTP status code and any possible data returned from the API can extracted from the `VimeoResponse.GenericError` class.
+
+
+## Certificate Pinning
+
+By default certificate pining is enabled for every request. If you would like to disable it, you may specify it when configuring the SDK. 
+
+```kotlin
+
+// Kotlin
+val certPinningEnabled = false
+val serverConfig = ServerConfig(CLIENT_ID, CLIENT_SECRET, scopes, certPinningEnabled)
+
+```
+
+If you are using the SDK on KitKat, then you will have to turn on certificate pinning. In KitKat, TLS is disabled by default. 
```

Comment:
Don't we have to disable certificate pinning on KitKat, not enable it? Maybe I am misremembering.

Reply:
Yeah. We have to turn it off. I had a typo.